src/textcheats.c

```diff
@@ -599,14 +599,12 @@ static inline int getToken(const unsigned char *line, const int len)
 
     if(len == 17 && line[8] == ' ')
     {
-        c = line;
+        c = line + numTokens;
```

Comment:
The search is performed from the beginning of the line, so `numTokens` doesn't need to be added.

Reply:
I did it as some valid codes were being incorrectly listed due to `getToken()` considering them as cheat lines (`TOKEN_CHEAT`) instead of codes (`TOKEN_CODE`).

In more detail, when a valid code line (e.g. `01234567 00000089`) is being processed at this point, `numTokens` (at line [602](/root670/CheatDevicePS2/blob/148ebc1f9ec9b06f0658e6dc53857a51c9f57e74/src/textcheats.c#L602)) is zero and `getToken()` returns `TOKEN_CODE` as the condition `(numTokens == 17)` (at line [606](/root670/CheatDevicePS2/blob/148ebc1f9ec9b06f0658e6dc53857a51c9f57e74/src/textcheats.c#L606)) is true, but in some cases, `numTokens` (at line [602](/root670/CheatDevicePS2/blob/148ebc1f9ec9b06f0658e6dc53857a51c9f57e74/src/textcheats.c#L602)) has a non-zero value (more specifically 8), and this causes that the condition `(numTokens == 17)` does not match due to `numTokens` being equal to 25, i.e. 8 plus the 17 counted tokens, so `getToken()` ends returning `TOKEN_CHEAT`.

In resume:

- Current:
```c
//At line 602
line = "01234567 00000089"
numTokens = 8
//At line 606
numTokens = 25
getToken() returns TOKEN_CHEAT
```

- Expected:
```c
//At line 602
line = "01234567 00000089"
numTokens = 0
//At line 606
numTokens = 17
getToken() returns TOKEN_CODE
```

Here is a small test case.
```
"Test Game"
//Comment line 1
Enable Code (Must Be On)
//Comment line 2
901969F8 0C0A3A26
Section #1
Cheat #1
//Comment line 3
D022DA9C 0000FFFE
2023FF78 24020001
D022DA9C 0010FFFE
2023FF78 8C620000
Cheat #2
00402D3F 00000018
Cheat #3
//Comment line 4
E002453B 1019597E
5019597C 00000004
00195978 00000000
Cheat #4
E001453B 1019597E
20195978 43480000
Cheat #5
//ThisIs AComment
20602D28 0000EA60
Cheat #6
00702D3F 00000018
```
Some screenshots:

- Test case being erroneously listed:
![bad_listed](https://user-images.githubusercontent.com/56147759/66250981-4ab33280-e718-11e9-9e6d-b3bc4cb76ab6.jpg)

- Expected test case listing:
![good_listed](https://user-images.githubusercontent.com/56147759/66250989-5bfc3f00-e718-11e9-8c2a-087b2425283e.jpg)

Reply:
Well, I believe that I have found the cause of this issue, it is related with the comments above code lines. Let me explain:

Suppose the following cheat:
```
Press Select To Skip FMV
//Warning: This Code Can Cause Some Nasty Side Effects
D022DA9C 0000FFFE
2023FF78 24020001
D022DA9C 0010FFFE
2023FF78 8C620000
```
Here we have a cheat with 4 code lines, so we expect to see the cheat title listed and no one of the code lines, but the comment between the cheat title and the 1st. code line is causing the misbehaviour.

```
//At parseLine() function
getToken("Press Select To Skip FMV") returns TOKEN_CHEAT
g_ctx.lastToken is set to TOKEN_CHEAT

... next line in cheat list ...
getToken("//Warning: This Code Can Cause Some Nasty Side Effects") returns 0
g_ctx.lastToken is set to 0

... next line in cheat list ...
getToken("D022DA9C 0000FFFE") returns TOKEN_CHEAT (expected: TOKEN_CODE)
```

See that `g_ctx.lastToken` zero value before `D022DA9C 0000FFFE` being processed at `getToken()`?. At line [580](/root670/CheatDevicePS2/blob/148ebc1f9ec9b06f0658e6dc53857a51c9f57e74/src/textcheats.c#L580) there is an `if` statement which evaluates to true when `g_ctx.lastToken == 0`. So, the code line gets processed here (although the last processed line was a comment), and at line [602](/root670/CheatDevicePS2/blob/148ebc1f9ec9b06f0658e6dc53857a51c9f57e74/src/textcheats.c#L602), `numTokens` is non-zero.

So, you stand corrected, there is no need to add `numTokens` at line 602, hence I must revert the change done at that line.

My proposed fix is removing the condition `g_ctx.lastToken == 0` from the `if` statement at line 580, so, the `if` statement should change from
```c
    if(UNLIKELY(len > 3 &&
       ((g_ctx.lastToken == TOKEN_MAP_START) || 
        (g_ctx.lastToken == TOKEN_MAP_ENTRY) ||
        (g_ctx.lastToken == 0))))
```
to
```c
    if(UNLIKELY(len > 3 &&
       ((g_ctx.lastToken == TOKEN_MAP_START) || 
        (g_ctx.lastToken == TOKEN_MAP_ENTRY))))
```

I am going to revert the change done at line 602 and apply the proposed fix.

Reply:
Thank you for finding this bug and giving such a thorough explanation. I think this shows that I need to create unit tests for this function since parsing issues can be difficult to diagnose.